aiidalab_widgets_base/viewers.py

```diff
@@ -1136,7 +1136,16 @@ def _observe_supercell(self, _=None):
             self.set_trait(
                 "displayed_structure", None
             )  # To make sure the structure is always updated.
-            self.set_trait("displayed_structure", self.structure.repeat(self.supercell))
+            # nglview displays structures by first saving them to a temporary "pdb" file, which necessitates
+            # converting the unit cell and atomic positions into a standard form where the a-axis aligns along the x-axis.
+            # This transformation can cause discrepancies between the atom positions and custom bonds calculated from the original structure.
+            # To mitigate this, we transform the "displayed_structure" into the standard form prior to rendering in nglview.
+            # This ensures that nglview's internal handling does not further modify the structure unexpectedly.
+            pdb_structure = self.structure.copy()
+            pdb_structure.set_cell(
+                self.structure.cell.standard_form()[0], scale_atoms=True
+            )
+            self.set_trait("displayed_structure", pdb_structure.repeat(self.supercell))
```

Comment:
```suggestion
            oriented_structure = self.structure.copy()
            oriented_structure.set_cell(
                self.structure.cell.standard_form()[0], scale_atoms=True
            )
            self.set_trait("displayed_structure", oriented_structure.repeat(self.supercell))
```

Reply:
thanks! I change it to `standard_structure`.